### DISCUSSION TITLE
Error: "= note: /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpython39: No such file or directory"

### DISCUSSION THREAD

**Discussion by otavioonoue** [PyO3/pyo3#3136] [Questions]
I don't know if this is the right place to ask questions about errors, however I need help on this point.

Executed command: `cargo build --target x86_64-pc-windows-gnu`

Error printscreen: 
![image](https://user-images.githubusercontent.com/34526678/236375493-8590abfc-a23b-4c81-9db1-bb2ce8ffba8c.png)

OS Information: 
![image](https://user-images.githubusercontent.com/34526678/236375682-6a2e5571-d6f5-4e26-a52e-d4a357d3dac5.png)

**Comment by messense** [comment]
Please provide error message in plaintext instead of screenshots.

How did you setup your mingw64 environment? Can you provide some code or git repository to reproduce your issue?

**Reply by otavioonoue** [reply]
Compiling pyo3teste v0.1.0 (/home/ddo/programação/pyo3teste)
error: linking with `x86_64-w64-mingw32-gcc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/ddo/.cargo/bin:/home/ddo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin" VSLANG="1033" "x86_64-w64-mingw32-gcc" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-m64" "-Wl,--high-entropy-va" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/rsbegin.o" "/tmp/rustcssNwnf/symbols.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.15dxvqtjumax2vyu.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1ge9jovtdi3kmznl.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1nj6v89vihjlyrk2.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1rtmjct4p84ywaj6.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1v8bvpb46qbg3ww0.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1xpfnsg2f18xm3tu.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2aoa259ifrs6uj2p.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2f6d3cim96mnpos6.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2pgq3w5m4n27d56i.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2vm0qpw7k7wnzx00.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.3mckqgr4ftv15c2z.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.42t7z3pm91jupefd.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.4e1vfcpw9pahla09.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.4jgxpqrml5hv0odg.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.4y7jz96pmhni5jvq.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.530q9or2mv8sbx14.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.53lrvvimunxwhnrm.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.dxllzd6sd10cbtp.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.vp6ys155sf0d7s7.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.vu2uf95kj7risq.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.xf5ymlo1iv6hjng.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2de8rjmhcrowye43.rcgu.o" "-L" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps" "-L" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/debug/deps" "-L" "/home/ddo/.cargo/registry/src/github.com-1ecc6299db9ec823/windows_x86_64_gnu-0.42.2/lib" "-L" "/usr/lib" "-L" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib" "-Wl,-Bstatic" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libpyo3-22880a9e55c01bfe.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libmemoffset-a3b412b80d9fd6ca.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libparking_lot-0ffbf71137517044.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libparking_lot_core-ec3e75564c908683.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libwindows_sys-fe56c29ec3ef3756.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libcfg_if-76c960a90ab5812e.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libsmallvec-3b2f700567261d63.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/liblock_api-a8089003dcef02f0.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libscopeguard-2d1d3a68f9b12390.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libpyo3_ffi-80165d21d0a75118.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/liblibc-8ffd19465750bcd4.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libunindent-1b8a149fa445f6f0.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libstd-4208b5b050761eab.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libpanic_unwind-c24e86db72ac50d0.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libobject-8130621e6418bc58.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libmemchr-08ae56106143c1db.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libaddr2line-ccfb784edb974a4a.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libgimli-4ab6c884d279676f.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_demangle-9ddceef41b9e9295.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libstd_detect-d8366e410ad5f9d7.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libhashbrown-ce69f37d51637f0b.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libminiz_oxide-19f6886d9ebc20bf.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libadler-25358c2b66d0aa3d.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_std_workspace_alloc-d18d5ff4d2dae958.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libunwind-0cef8242de7339bc.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcfg_if-edcd03acb0b5d165.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/liblibc-21b2a42330a7f2e3.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/liballoc-ee725b426657593b.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_std_workspace_core-d6f1a8662e8b71a9.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcore-84ddc19bfcf64c94.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcompiler_builtins-6b61207aa476e831.rlib" "-Wl,-Bdynamic" "-lwindows" "-lpython39" "-lkernel32" "-ladvapi32" "-luserenv" "-lkernel32" "-lws2_32" "-lbcrypt" "-lgcc_eh" "-l:libpthread.a" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib" "-o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.exe" "-Wl,--gc-sections" "-no-pie" "-nodefaultlibs" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/rsend.o"
  = note: /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpython39: No such file or directory
          collect2: error: ld returned 1 exit status
          

error: could not compile `pyo3teste` due to previous error

**Reply by adamreichold** [reply]
How did you obtain your Windows build of Python and how did you tell PyO3 where to find it?

If you do not have a Windows build of Python then the error is that there is no such build and hence we cannot link against a Windows version the Python library. You might want to consider enabling the [`abi3-py39`](https://pyo3.rs/v0.18.3/features#the-abi3-pyxy-features) and [`generate-import-lib`](https://pyo3.rs/v0.18.3/features#generate-import-lib) features, e.g.

```toml
pyo3 = { version = "0.18", features = ["abi3-py39", "generate-import-lib"] }
```

to target a configuration that does not need a Windows build available.

**Reply by otavioonoue** [reply]
`How did you setup your mingw64 environment?`: Could you provide me with more information about what you want to know? I did this setup a long time ago and I don't remember exactly what I did.

`Can you provide some code or git repository to reproduce your issue? `:  For sure: https://github.com/Ott-cop/pyo3-test

**Reply by otavioonoue** [reply]
> How did you obtain your Windows build of Python and how did you tell PyO3 where to find it?
> 
> If you do not have a Windows build of Python then the error is that there is not such build and hence we cannot link against a Windows version the Python library. You might want to consider enabling the [`abi3-py39`](https://pyo3.rs/v0.18.3/features#the-abi3-pyxy-features) and [`generate-import-lib`](https://pyo3.rs/v0.18.3/features#generate-import-lib) features, e.g.
> 
> ```toml
> pyo3 = { version = "0.18", features = ["abi3-py39", "generate-import-lib"] }
> ```
> 
> to target a configuration that does not need a Windows build available.

I made the changes and apparently got the same error: 
error: linking with `x86_64-w64-mingw32-gcc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/home/ddo/.cargo/bin:/home/ddo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin" VSLANG="1033" "x86_64-w64-mingw32-gcc" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-m64" "-Wl,--high-entropy-va" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/rsbegin.o" "/tmp/rustcuyzW5E/symbols.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.15dxvqtjumax2vyu.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1ge9jovtdi3kmznl.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1nj6v89vihjlyrk2.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1rtmjct4p84ywaj6.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1v8bvpb46qbg3ww0.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.1xpfnsg2f18xm3tu.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2aoa259ifrs6uj2p.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2f6d3cim96mnpos6.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2pgq3w5m4n27d56i.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2vm0qpw7k7wnzx00.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.3mckqgr4ftv15c2z.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.42t7z3pm91jupefd.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.4e1vfcpw9pahla09.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.4jgxpqrml5hv0odg.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.4y7jz96pmhni5jvq.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.530q9or2mv8sbx14.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.53lrvvimunxwhnrm.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.dxllzd6sd10cbtp.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.vp6ys155sf0d7s7.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.vu2uf95kj7risq.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.xf5ymlo1iv6hjng.rcgu.o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.2de8rjmhcrowye43.rcgu.o" "-L" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps" "-L" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/debug/deps" "-L" "/home/ddo/.cargo/registry/src/github.com-1ecc6299db9ec823/windows_x86_64_gnu-0.42.2/lib" "-L" "/usr/lib" "-L" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib" "-Wl,-Bstatic" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libpyo3-22880a9e55c01bfe.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libmemoffset-a3b412b80d9fd6ca.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libparking_lot-0ffbf71137517044.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libparking_lot_core-ec3e75564c908683.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libwindows_sys-fe56c29ec3ef3756.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libcfg_if-76c960a90ab5812e.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libsmallvec-3b2f700567261d63.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/liblock_api-a8089003dcef02f0.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libscopeguard-2d1d3a68f9b12390.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libpyo3_ffi-80165d21d0a75118.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/liblibc-8ffd19465750bcd4.rlib" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/libunindent-1b8a149fa445f6f0.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libstd-4208b5b050761eab.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libpanic_unwind-c24e86db72ac50d0.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libobject-8130621e6418bc58.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libmemchr-08ae56106143c1db.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libaddr2line-ccfb784edb974a4a.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libgimli-4ab6c884d279676f.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_demangle-9ddceef41b9e9295.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libstd_detect-d8366e410ad5f9d7.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libhashbrown-ce69f37d51637f0b.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libminiz_oxide-19f6886d9ebc20bf.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libadler-25358c2b66d0aa3d.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_std_workspace_alloc-d18d5ff4d2dae958.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libunwind-0cef8242de7339bc.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcfg_if-edcd03acb0b5d165.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/liblibc-21b2a42330a7f2e3.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/liballoc-ee725b426657593b.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/librustc_std_workspace_core-d6f1a8662e8b71a9.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcore-84ddc19bfcf64c94.rlib" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/libcompiler_builtins-6b61207aa476e831.rlib" "-Wl,-Bdynamic" "-lwindows" "-lpython39" "-lkernel32" "-ladvapi32" "-luserenv" "-lkernel32" "-lws2_32" "-lbcrypt" "-lgcc_eh" "-l:libpthread.a" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib" "-o" "/home/ddo/programa\xc3\xa7\xc3\xa3o/pyo3teste/target/x86_64-pc-windows-gnu/debug/deps/pyo3teste-cbd43cdc5fd4746c.exe" "-Wl,--gc-sections" "-no-pie" "-nodefaultlibs" "/home/ddo/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-pc-windows-gnu/lib/rsend.o"
  = note: /usr/bin/x86_64-w64-mingw32-ld: cannot find -lpython39: No such file or directory
          collect2: error: ld returned 1 exit status
          

error: could not compile `pyo3teste` due to previous error

**Reply by otavioonoue** [reply]
`How did you obtain your Windows build of Python and how did you tell PyO3 where to find it?`: I'm not sure, I made several attempts, internet searches trying to redo it to see if it would, but nothing resolved

**Reply by otavioonoue** [reply]
I'm a beginner so excuse anything
